### PR TITLE
Joomla 4.2.0

### PR DESCRIPTION
**Pull request by roland-d**
4.2.0 stable release

@zero-24 Could you please check if this is OK or garbage? :)

### CODE REVIEW

File: www/core/nightlies/next_patch_list.xml [hunk 1]
```diff
@@ -5,4 +5,5 @@
 	<extension name="Joomla" element="joomla" type="file" version="3.10.11-dev" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.6-dev" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.6-dev" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="4.2.1-dev" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
 </extensionset>
```
Comment (zero-24) [inline www/core/nightlies/next_patch_list.xml:9]: 4.2 line is missing

File: www/core/nightlies/next_patch_list.xml [hunk 2]
```diff
@@ -5,4 +5,5 @@
 	<extension name="Joomla" element="joomla" type="file" version="3.10.11-dev" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.6-dev" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.6-dev" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="4.2.1-dev" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
```
Comment (zero-24) [inline www/core/nightlies/next_patch_list.xml:8]: everyline has to be updated to 4.2.1-dev

File: www/core/sts/extension_sts.xml [hunk 3]
```diff
@@ -20,50 +20,50 @@
...
 		<php_minimum>7.2.5</php_minimum>
-		<sha256>4ca1dcb9a603ceaa851a233c5232ace618239c5348d89b89966b47830e12f0b7</sha256>
-		<sha384>ece47292198066d2d9049d23175e4635857f326262b3a581196817086c723e631296bf30d2791999e171ddd7009b57ea</sha384>
-		<sha512>7d05270e83420ccd76b1f463d485358affb21357cc26e23a1ea18161cf2b354ccf601e80c64abce509280a052e1154f84a8c802bbcb7f4355ae35fe5b32921d2</sha512>
+		<sha256>TBD</sha256>
+		<sha384>TBD</sha384>
+		<sha512>TBD</sha512>
 		<maintainer>Joomla! Production Department</maintainer>
 		<maintainerurl>https://www.joomla.org</maintainerurl>
 		<section>STS</section>
```
Comment (zero-24) [inline www/core/sts/extension_sts.xml:69]: just below this line the target platform change is missing

File: www/core/sts/list_sts.xml [hunk 4]
```diff
@@ -14,7 +14,10 @@
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.8" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.9" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="3.10.10" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="4.2.0" targetplatformversion="3.10.10" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
```
Comment (zero-24) [inline www/core/sts/list_sts.xml:20]: can be removed as the new lines for 4.0 and 4.1 is already there.

File: www/core/sts/list_sts.xml [hunk 5]
```diff
@@ -14,7 +14,10 @@
...
+	<extension name="Joomla" element="joomla" type="file" version="4.2.0" targetplatformversion="3.10.10" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="4.2.0" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="4.2.0" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 </extensionset>
```
Comment (zero-24) [inline www/core/sts/list_sts.xml:23]: a new targetplatformversion="4.2" is missing.

File: www/core/nightlies/next_patch_extension.xml [hunk 6]
```diff
@@ -18,14 +18,14 @@
...
 		<type>file</type>
-		<version>4.1.6-dev</version>
+		<version>4.2.1-dev</version>
 		<infourl title="Joomla! Nightly Builds">https://developer.joomla.org/nightly-builds.html</infourl>
 		<downloads>
-			<downloadurl type="full" format="zip">https://developer.joomla.org/nightlies/Joomla_4.1.6-dev-Development-Update_Package.zip</downloadurl>
+			<downloadurl type="full" format="zip">https://developer.joomla.org/nightlies/Joomla_4.2.1-dev-Development-Update_Package.zip</downloadurl>
 		</downloads>
 		<tags>
 			<tag>stable</tag>
```
Comment (zero-24) [inline www/core/nightlies/next_patch_extension.xml:31]: the targetplatform version has to be adjusted (a few lines below this comment)

File: www/core/list.xml [hunk 7]
```diff
@@ -16,4 +16,7 @@
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="4.1.5" targetplatformversion="4.1" detailsurl="https://update.joomla.org/core/sts/extension_sts.xml" />
```
Comment (zero-24) [inline www/core/list.xml:18]: the 4.1.5 lines should be removed from here.

**Review by zero-24 — COMMENTED** [review]
I have did some comments here. We also miss the changes for 3.10.11 which is usually packed together within one PR so it does not conflict and can be merged as one

**Comment by roland-d** [comment]
> I have did some comments here. We also miss the changes for 3.10.11 which is usually packed together within one PR so it does not conflict and can be merged as one

I fixed your comments, hope it is OK now. Do you want me to add the 3.10.11 as well or will you send a PR against this one?

**Comment by zero-24** [comment]
Please add the changes for 3.10 just leave the hashes blank. For how it should look like take a look here: https://github.com/joomla/update.joomla.org/pull/251/files

**Comment by roland-d** [comment]
@zero-24 I have added 3.10.11 as well, can you check if it is OK? Thanks.

File: www/core/nightlies/next_patch_list.xml [hunk 8]
```diff
@@ -1,8 +1,9 @@
...
-	<extension name="Joomla" element="joomla" type="file" version="3.10.11-dev" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="3.10.12-dev" targetplatformversion="3.7" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="3.10.12-dev" targetplatformversion="3.8" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="3.10.12-dev" targetplatformversion="3.9" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="3.10.12-dev" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
+	<extension name="Joomla" element="joomla" type="file" version="3.10.12-dev" targetplatformversion="3.11" detailsurl="https://update.joomla.org/core/nightlies/next_patch_extension.xml" />
```
Comment (zero-24) [inline www/core/nightlies/next_patch_list.xml:6]: There is no 3.11?

File: www/core/test/extension_test.xml [hunk 9]
```diff
@@ -5,10 +5,10 @@
 		<description>Joomla! 3.10 CMS</description>
 		<element>joomla</element>
 		<type>file</type>
-		<version>3.10.10</version>
-		<infourl title="Joomla 3.10.10 Release">https://github.com/joomla/joomla-cms/releases/tag/3.10.10</infourl>
+		<version>3.10.11</version>
+		<infourl title="Joomla 3.10.11 Release">https://github.com/joomla/joomla-cms/releases/tag/3.10.11</infourl>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:9]: This should be pointing to the final release URL on joomla.org

File: www/core/test/extension_test.xml [hunk 10]
```diff
@@ -23,17 +23,17 @@
...
```
Comment (zero-24) [inline www/core/test/extension_test.xml:27]: Also this should go to the finale release URL please

**Review by zero-24 — CHANGES_REQUESTED** [review]
added some comments and also found that the 3.10 changes to the following files are missing:
www/core/list.xml - missing
www/core/extension.xml - missing
www/core/sts/list_sts.xml - missing

**Comment by zero-24** [comment]
This also conflicted as I have just merged the 3.10.11-rc1 changes.

File: www/core/nightlies/next_patch_list.xml (hunk 8 above)
Comment (roland-d) [reply www/core/nightlies/next_patch_list.xml:6]: You know I have no idea what I am doing. Just guessing what is correct.

File: www/core/test/extension_test.xml (hunk 9 above)
Comment (roland-d) [reply www/core/test/extension_test.xml:9]: If the old one is for 3.10.10, should it not be the same for 3.10.11?

File: www/core/test/extension_test.xml (hunk 10 above)
Comment (roland-d) [reply www/core/test/extension_test.xml:27]: Is the final release URL different for 3.10.11 compared to 3.10.10?

File: www/core/nightlies/next_patch_list.xml (hunk 8 above)
Comment (zero-24) [reply www/core/nightlies/next_patch_list.xml]: resolved

File: www/core/sts/extension_sts.xml [hunk 11]
```diff
@@ -20,53 +20,53 @@
...
-		<name>Joomla! 4.1</name>
-		<description>Joomla! 4.1 CMS</description>
+		<name>Joomla! 4.2</name>
+		<description>Joomla! 4.2 CMS</description>
 		<element>joomla</element>
 		<type>file</type>
-		<version>4.1.5</version>
-		<infourl title="Joomla 4.1.5 Release">https://www.joomla.org/announcements/release-news/5861-joomla-4-1-5-and-3-10-10-release.html</infourl>
+		<version>4.2.0</version>
+		<infourl title="Joomla 4.2.0 Release">TBD</infourl>
```
Comment (zero-24) [inline www/core/sts/extension_sts.xml:28]:
```suggestion
		<infourl title="Joomla 4.2.0 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>
```

File: www/core/sts/extension_sts.xml [hunk 12]
```diff
@@ -20,53 +20,53 @@
...
```
Comment (zero-24) [inline www/core/sts/extension_sts.xml:53]:
```suggestion
		<infourl title="Joomla 4.2.0 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>

```

File: www/core/test/310to4_extension.xml [hunk 13]
```diff
@@ -5,19 +5,21 @@
 		<description>Joomla! 4.2 CMS</description>
 		<element>joomla</element>
 		<type>file</type>
-		<version>4.2.0-rc1</version>
-		<infourl title="Joomla 4.2.0-rc1 Release">https://www.joomla.org/announcements/release-news/5862-joomla-4-2-release-candidate-1-help-make-it-stable.html</infourl>
+		<version>4.2.0</version>
+		<infourl title="Joomla 4.2.0 Release">https://www.joomla.org/announcements/release-news/5862-joomla-4-2-release-candidate-1-help-make-it-stable.html</infourl>
```
Comment (zero-24) [inline www/core/test/310to4_extension.xml:9]:
```suggestion
		<infourl title="Joomla 4.2.0 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>

```

File: www/core/test/extension_test.xml (hunk 9 above)
Comment (zero-24) [reply www/core/test/extension_test.xml:9]:
```suggestion
		<infourl title="Joomla 3.10.11 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>
```

File: www/core/test/extension_test.xml [hunk 14]
```diff
@@ -23,17 +23,17 @@
 		<description>Joomla! 3.10 CMS</description>
 		<element>joomla</element>
 		<type>file</type>
-		<version>3.10.11-rc1</version>
-		<infourl title="Joomla 3.10.11-rc1 Release">https://github.com/joomla/joomla-cms/releases/tag/3.10.11-rc1</infourl>
+		<version>3.10.11</version>
+		<infourl title="Joomla 3.10.11 Release">https://github.com/joomla/joomla-cms/releases/tag/3.10.11</infourl>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:27]:
```suggestion
		<infourl title="Joomla 3.10.11 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>
```

File: www/core/test/extension_test.xml [hunk 15]
```diff
@@ -44,17 +44,17 @@
 		<description>Joomla! 4.2 CMS</description>
 		<element>joomla</element>
 		<type>file</type>
-		<version>4.2.0-rc1</version>
-		<infourl title="Joomla 4.2.0-rc1 Release">https://github.com/joomla/joomla-cms/releases/tag/4.2.0-rc1</infourl>
+		<version>4.2.0</version>
+		<infourl title="Joomla 4.2.0 Release">https://github.com/joomla/joomla-cms/releases/tag/4.2.0</infourl>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:48]:
```suggestion
		<infourl title="Joomla 4.2.0 Release">https://www.joomla.org/announcements/release-news/5865-joomla-4-2-release.html</infourl>
```

File: www/core/test/extension_test.xml [hunk 16]
```diff
@@ -23,17 +23,17 @@
...
 		</downloads>
 		<tags>
 			<tag>stable</tag>
 		</tags>
-		<sha256>1a4fd2f0d2fc87d29fe095a6e85d4a2ceec9636a2836e0376a6192039d15ef53</sha256>
-		<sha384>bf49dff154ca37b2c58a211689823069529dfd33f4a703998c297ec97e89bcc05785558130036342dd932c17ee4141a6</sha384>
-		<sha512>7b285e87855f51617c37579fd2c74afedfafc962f68d525c6468fc566077a3ceb1ccccc62c418a17e493d8dc752ba3dea82c6a265e7d35fdf67f62ed9e9afa76</sha512>
+		<sha256>TBD</sha256>
+		<sha384>TBD</sha384>
+		<sha512>TBD</sha512>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:36]:
```suggestion
		<sha512>742c8ba844e9f8d131a9a67202c77251dadab1664c84c161a8d1c2ac5503ac23be4ee7f73b2da7078ad0dccb8b70e1dd8e49ea0871cad78a6eb0da00ff29e505</sha512>
```

File: www/core/test/extension_test.xml [hunk 17]
```diff
@@ -23,17 +23,17 @@
...
-			<downloadurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.10.11-rc1/Joomla_3.10.11-rc1-Release_Candidate-Update_Package.zip</downloadurl>
+			<downloadurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.10.10/Joomla_3.10.11-Stable-Update_Package.zip</downloadurl>
 		</downloads>
 		<tags>
 			<tag>stable</tag>
 		</tags>
-		<sha256>1a4fd2f0d2fc87d29fe095a6e85d4a2ceec9636a2836e0376a6192039d15ef53</sha256>
-		<sha384>bf49dff154ca37b2c58a211689823069529dfd33f4a703998c297ec97e89bcc05785558130036342dd932c17ee4141a6</sha384>
-		<sha512>7b285e87855f51617c37579fd2c74afedfafc962f68d525c6468fc566077a3ceb1ccccc62c418a17e493d8dc752ba3dea82c6a265e7d35fdf67f62ed9e9afa76</sha512>
+		<sha256>TBD</sha256>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:34]:
```suggestion
		<sha256>12e0fe45ee53eedd776d3b062cda1cd8c38ce974f74e7f92eb1bad2e96ddabcd</sha256>
```

File: www/core/test/extension_test.xml [hunk 18]
```diff
@@ -23,17 +23,17 @@
...
+		<sha384>TBD</sha384>
```
Comment (zero-24) [inline www/core/test/extension_test.xml:35]:
```suggestion
		<sha384>93fa388435fd39e93d261216e8bb745139262a804ec45e9211aecd9086d15be73553ac288391237e4774ad9532adf655</sha384>
```

File: www/core/list.xml [hunk 19]
```diff
@@ -14,6 +14,7 @@
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.8" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.9" detailsurl="https://update.joomla.org/core/extension.xml" />
 	<extension name="Joomla" element="joomla" type="file" version="3.10.10" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/extension.xml" />
```
Comment (zero-24) [inline www/core/list.xml:16]:
```suggestion
	<extension name="Joomla" element="joomla" type="file" version="3.10.11" targetplatformversion="3.10" detailsurl="https://update.joomla.org/core/extension.xml" />

```

**Pull request merged**